### PR TITLE
Support Pitch, Rate and Volume commands properly in SAPI4

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None

### Summary of the issue:
Pitch, volume and rate changes can be embedded in speech sequences. However, the latter two were never implemented for SAPI4. Pitch support was implemented last year, but it was pretty strictly bound to the capital pitch change setting (i.e. it didn't respect the offset on the PitchCommand.

### Description of user facing changes
Capital pitch change intervals might be bigger with some engines.

### Description of development approach
Pitch, Volume and Rate commands are now implemented according to the respective [control tags](http://www.tapiex.com/ES_Help/SctrlTags.htm)

### Testing strategy:
Tested that capital pitch change still works, though the difference in pitch might be bigger now. However, I compared the capital pitch change with a real pitch change and the intervals are equal.

### Known issues with pull request:
I'm not sure whether all engines support the control tags out there, but since SAPI4 is pretty legacy, we should stop bothering at some point and we should consider implementing exceptions in add-ons, especially now there is an add-on store.

### Change log entries:
Bug fixes
- The SAPI4 synthesizer now properly supports volume, rate and pitch changes embedded in speech.

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by leonardder** [comment]
Cc @cary-rowen, @dawidpieper

File: source/synthDrivers/sapi4.py [hunk 1]
```diff
@@ -239,8 +242,12 @@ def _set_voice(self,val):
 		if hasRate:
 			if not self.isSupported('rate'):
 				self.supportedSettings.insert(1,SynthDriver.RateSetting())
+			self.supportedCommands.add(RateCommand)
 		else:
-			if self.isSupported("rate"): self.removeSetting("rate")
+			if self.isSupported("rate"):
+				self.removeSetting("rate")
+		if RateCommand in self.supportedCommands:
+			self.supportedCommands.remove(RateCommand)
```
Comment (seanbudd) [inline source/synthDrivers/sapi4.py:250]: should this be only if `hasRate == False`?

Comment (seanbudd) [reply source/synthDrivers/sapi4.py:250]:
```suggestion
			if RateCommand in self.supportedCommands:
				self.supportedCommands.remove(RateCommand)
```

File: source/synthDrivers/sapi4.py [hunk 2]
```diff
@@ -304,29 +318,32 @@ def _getAvailableVoices(self):
...
 	def _get_rate(self):
-		val=DWORD()
+		val = DWORD()
 		self._ttsAttrs.SpeedGet(byref(val))
-		return self._paramToPercent(val.value,self._minRate,self._maxRate)
+		return self._paramToPercent(val.value, self._minRate, self._maxRate)
 
-	def _set_rate(self,val):
-		val=self._percentToParam(val,self._minRate,self._maxRate)
+	def _set_rate(self, val):
```
Comment (seanbudd) [inline source/synthDrivers/sapi4.py:325]: could the setter and getter functions get type annotations added?

**Review by seanbudd — APPROVED** [review]

**Comment by codeofdusk** [comment]
The changelog entry for this is under 2023.3, not 2024.1. Is this intentional?

**Pull request merged**